### ISSUE TITLE
[DocDB] Fast PITR - Set cotables filter during PITR

### ISSUE THREAD

**Issue by sanketkedia** [yugabyte/yugabyte-db#17464]
Jira Link: [DB-6629](https://yugabyte.atlassian.net/browse/DB-6629)
### Description

After #17270, we want to leverage its infrastructure to correctly set cotable filter for the restored database during PITR. We introduce a new flag `enable_fast_pitr` which when set to true uses the new faster PITR workflow that involves the following:

1. Open rocksdb on the side for the snapshot, iterate over the current state and restoring state and compute the differences between them for two tables - the `docdb metadata aka legacy sys catalog table` and the `pg_yb_migration` table. Also, obtain the current catalog version number from `pg_yb_catalog_version` table.
2. Error out if the `pg_yb_migration` table has changed. This blocks going back to a time before a ysql_upgrade which is not supported.
3. Otherwise, generate a write batch with these differences as well as the incremented catalog version
4. Flush and shutdown rocksdb, write the cotable filter in the MANIFEST for all the SST files and start rocksdb again.
5. Write the write batch generated in (3)

Please note that this issue does not ensure that the entire relevant history is retained for the pg catalog tables in the current active rocksdb, that is another issue which will be followed up next.

### Warning: Please confirm that this issue does not contain any sensitive information

- [X] I confirm this issue does not contain any sensitive information.

[DB-6629]: https://yugabyte.atlassian.net/browse/DB-6629?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
